jsm.go

```diff
@@ -244,13 +264,10 @@ func (js *js) AccountInfo(opts ...JSOpt) (*AccountInfo, error) {
 		return nil, err
 	}
 	if info.Error != nil {
-		var err error
-		if strings.Contains(info.Error.Description, "not enabled for") {
-			err = ErrJetStreamNotEnabled
-		} else {
-			err = errors.New(info.Error.Description)
+		if info.Error.ErrorCode == JetStreamNotEnabledForAccount {
+			return nil, ErrJetStreamNotEnabled
```

Comment:
In the server not enabled for account and not enabled looks like they are two different errors actually... maybe we should start distinguishing them too and use `ErrJetStreamNotEnabledForAccount`?

https://github.com/nats-io/nats-server/blob/52c4872666a26acdf475cb89ae9b5f367324fadd/server/jetstream_errors_generated.go#L472-L474

Reply:
Another factor why maybe we should clean it up is that `JetStreamNotEnabledForAccount` is a hard error since means that account can never use JetStream in current setup, whereas things are a bit less certain in `ErrJetStreamNotEnabled`, it can be either a temporary error due to JS not ready yet.

Reply:
Yes, distinguishing these errors is a good idea.